R/OsModel.R

```diff
@@ -0,0 +1,171 @@
+                              parameters = "os_parameters.stan",
+                              transformed_parameters = "os_transformed_parameters.stan",
+                              generated_quantities = "os_generated_quantities.stan",
+                              priors = os_prior(),
+                              inits = list(),
+                              templated = TRUE) {
+    st_mod <- StanModule(
+            functions = functions,
+            data = data,
+            parameters = parameters,
+            transformed_parameters = transformed_parameters,
+            generated_quantities = generated_quantities,
+            priors = priors,
+            inits = inits
+        )
+
+    OsModel(
+        stan = st_mod,
+        templated = templated
+    )
+}
+
+
+
+
+
+
+#' LogLogisticOs object creator
+#'
+#' Creates a log-logistic overall survival templated object
+#' @rdname LogLogisticOs-class
+#' @importFrom assertthat assert_that
+#' @export
+setMethod(
+    f = "initialize",
+    signature = "LogLogisticOs",
+    definition = function(.Object,
+                          ..., stan = LogLogisticModule(), templated) {
+        callNextMethod(
+            .Object,
+            ...,
+            stan = stan@stan,
+            templated = stan@templated
+        )
+    }
+)
+
+
+
+
+
+
+
+
+#' Parametrize TemplatedStanOs object with the selected Hazardlink
+#' @param osmod TemplatedStanOs object
+#' @param link HazardLink object
+#' @importFrom stringr str_replace
+#' @importFrom stringr str_remove_all
+#' @importFrom stringr str_replace_all
+#' @export
+setGeneric("parametrize", function(osmod, link) {
+    standardGeneric("parametrize")
+})
+
+
+
+
+
+# To be updated
+setMethod(
+    "parametrize",
+    signature(osmod = "OsModel", link = "HazardLink"),
+    function(osmod, link) {
+        newOS <- OsModel(stan = StanModule(
+            functions = osmod@stan@functions,
+            data = osmod@stan@data,
+            parameters = osmod@stan@data,
+            transformed_parameters = osmod@stan@transformed_parameters,
+            priors = osmod@stan@priors,
+            generated_quantities = osmod@stan@generated_quantities,
+            inits = osmod@stan@inits
+        ))
+
+        newOS@stan@functions <- str_replace_all(
+            string = osmod@stan@functions,
+            pattern = "<link_arguments>",
+            replacement = paste0("real ", link@parameters, ",")
+        ) |>
+            paste0("\\n ", link@stan@functions) |>
+            str_replace(
+                pattern = "<link_log_hazard_contribution>",
+                replacement = paste0(link@parameters, link@contribution)
+            ) |>
+            str_replace(
+                pattern = "<link_arguments_as_par>",
+                replacement = paste0(link@parameters)
+            )
+
+
+        newOS@stan@priors <- append(osmod@stan@priors, link@stan@priors)
+        newOS@stan@inits <- append(osmod@stan@inits, link@stan@inits)
```

Comment:
Is this not combination not handled by the merge method ?

Reply:
We can append the priors and the initial values lists using the merge method as well. I will implement it this way.